test/test_main.cpp

```diff
@@ -1914,3 +1914,50 @@ TEST_F(ConnectionParsing, RequestUriTooLarge) {
     ASSERT_EQ(HTP_REQUEST_COMPLETE, tx->request_progress);
     ASSERT_EQ(HTP_RESPONSE_COMPLETE, tx->response_progress);
 }
+
+
+
+static std::string collected_req;
+static int collect_raw_reqest(htp_tx_data_t *d) {
+    printf("reqc: %d\n", (int)d->len);
+    collected_req += std::string((const char*)d->data, d->len);
+    return HTP_OK;
+}
+
+static std::string collected_resp;
+static int collect_raw_response(htp_tx_data_t *d) {
+    printf("respc: %d\n", (int)d->len);
+    collected_resp += std::string((const char*)d->data, d->len);
+    return HTP_OK;
+}
+
+
+TEST_F(ConnectionParsing, RawBodyHooks) {
+
+    collected_req.clear();
+    collected_resp.clear();
+    htp_config_register_request_raw_body_data(cfg, collect_raw_reqest);
+    htp_config_register_response_raw_body_data(cfg, collect_raw_response);
+    htp_config_register_urlencoded_parser(cfg);
+
+    int rc = test_run(home, "91-both-body-chunked.t", cfg, &connp);
+    ASSERT_GE(rc, 0);
+
+    ASSERT_EQ(1, htp_list_size(connp->conn->transactions));
+
+    htp_tx_t *tx = (htp_tx_t *) htp_list_get(connp->conn->transactions, 0);
+    ASSERT_TRUE(tx != NULL);
+
+    ASSERT_EQ(HTP_REQUEST_COMPLETE, tx->request_progress);
+    ASSERT_EQ(HTP_RESPONSE_COMPLETE, tx->response_progress);
+
+    printf("req: %s\n", collected_req.c_str());
```

Comment:
these `printf`s are not for complete code. but for debugging. So, callback are not fired. What I'm doing wrong?